src/plugins/workspace/server/saved_objects/integration_tests/workspace_id_consumer_wrapper.test.ts

```diff
@@ -140,6 +159,79 @@ describe('workspace_id_consumer integration test', () => {
       );
     });
 
+    it('bulk create with disallowed types in workspace', async () => {
+      await clearFooAndBar();
+
+      // import advanced settings and data sources should throw error
+      const createResultFoo = await osdTestServer.request
+        .post(root, `/w/${createdFooWorkspace.id}/api/saved_objects/_bulk_create`)
+        .send([
+          {
+            ...dataSource,
+            id: 'foo',
+          },
+          {
+            ...advancedSettings,
+            id: packageInfo.version,
+          },
+        ])
+        .expect(200);
+      expect(createResultFoo.body.saved_objects[0].error).toEqual(
+        expect.objectContaining({
+          message:
+            "Unsupport type in workspace: 'data-source' is not allowed to import in workspace.",
+          statusCode: 400,
+        })
+      );
+      expect(createResultFoo.body.saved_objects[1].error).toEqual(
+        expect.objectContaining({
+          message: "Unsupport type in workspace: 'config' is not allowed to import in workspace.",
+          statusCode: 400,
+        })
+      );
+
+      // data source and advanced settings should not be found within the workspace
```

Comment:
Does it means when create index pattern, there is no way to select datasource?

Reply:
This comment means the data source should not be created inside the workspace and thus can not be found. Let me adjust this comment a little bit.